### ISSUE TITLE
QPT NullPointerException(s)

### ISSUE THREAD

**Issue by swalker2m** [gemini-hlsw/ocs#1497]
A number of us, very much including myself, have modified the QPT over the years without really knowing what we're doing.  Likely as a consequence, it's pretty easy to see NPEs scroll by as you use the QPT.  I can't detect any issue in the UI itself as a result, but it is disconcerting.

The issue appears to be assuming a model is available when it isn't for some reason.  For example in the `CandidateObsController`:

```java
	public synchronized Object getSubElement(Obs obs, CandidateObsAttribute subElement) {
		switch (subElement) {
...
		case RA: return obs.getRa(viewer.getModel().getMiddlePoint());
...
		}
		return null;
	}
```

Getting the RA will sometimes throw a NPE because `viewer.getModel()` is `null`.

Other examples in `Visualizer`:

```java
	private Shape getMoonCurve(long start, long end) {
		synchronized (cacheLock) {

			// Try to use the cached version if we can.
...
			// Cache lookup failed, oh well. Calculate the moon's path.
			ImprovedSkyCalc calc = new ImprovedSkyCalc(model.getSchedule().getSite());
```

sometimes is surprised to find a `null` model.  Similarly in `getElevationCurve`:

```java
	private Shape getElevationCurve(Function<Long, WorldCoords> coords, long start, long end, boolean close, String obsId, Map<String, CachedShape> cache) {
		synchronized (cacheLock) {

			// Return the cached shape, if any.
...
			// Otherwise build the shape.
			ImprovedSkyCalc calc = new ImprovedSkyCalc(model.getSchedule().getSite());
```

Since the `model` may not be there when you need it, most of these methods could be edited to return an `Option<Foo>` but perhaps the real issue is that the model should always be there.

**Comment by tpolecat** [comment]
It's been a while but I think the model is null when no plan is open, and may be momentarily null when refreshing from the database. Are you seeing this when there's definitely a plan open?

**Comment by swalker2m** [comment]
> Are you seeing this when there's definitely a plan open?

Yes, on refresh.

**Comment by tpolecat** [comment]
Yep, ok. In any case you're right about using `Option`. Assuming our glorious future includes QPT in its current form we will need to put in some effort modernizing the client code.
